File: tests/CaseStmt.pas
namespace N;

type
  TTest = public class
  public
    method Foo(x: Integer);
  end;

  EnumCase = public class
  public
    method WeekDay(day: DayOfWeek);
  end;

implementation

method TTest.Foo(x: Integer);
begin
  case x of
    1: exit;
    2: exit;
  end;
end;

method EnumCase.WeekDay(day: DayOfWeek);
begin
  case day of
    DayOfWeek.Friday: System.Console.WriteLine('fri');
    DayOfWeek.Monday: System.Console.WriteLine('mon');
  end;
end;

end.

File: tests/CaseStmt.cs
namespace N {
    public partial class TTest {
        public void Foo(int x) {
            switch (x)
            {
                case 1: return; break;
                case 2: return; break;
            }
        }
    }
    
    public partial class EnumCase {
        public void WeekDay(DayOfWeek day) {
            switch (day)
            {
                case DayOfWeek.Friday: System.Console.WriteLine("fri"); break;
                case DayOfWeek.Monday: System.Console.WriteLine("mon"); break;
            }
        }
    }
}
